src/app/components/search/search.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, OnInit } from '@angular/core';
+import { Search } from 'src/app/models/Search';
+import { FormControl, FormGroup } from '@angular/forms';
+import { NgxSpinnerService } from 'ngx-spinner';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-search',
+  templateUrl: './search.component.html',
+  styleUrls: ['./search.component.scss']
+})
+export class SearchComponent implements OnInit {
+  searchResults: Search[] = [];
+  searchForm = new FormGroup({
+    title: new FormControl(),
+    description: new FormControl(),
+    street: new FormControl(),
+    state: new FormControl(),
+    company: new FormControl(),
+    city: new FormControl(),
+    type: new FormControl(),
+    purchase_plan: new FormControl(),
+    garages: new FormControl(),
+    bathrooms: new FormControl(),
+    bedrooms: new FormControl(),
+    lot_size: new FormControl(),
+    price_min: new FormControl(),
+    price_max: new FormControl(),
+  })
+
+  constructor(
+    private spinner: NgxSpinnerService,
+    private router: Router,
+  ) { }
+
+  ngOnInit() { }
+  
+  search({ value }) {
+    this.spinner.show();
+    let urlValues: string = '';
```

Comment:
I was thinking that this can be re-written to
```js
let urlValues = Object.keys(value).reduce((acc, key) => value[key] ? acc + key + '=' + value[key] + '&' : acc, "")
```

my reason is when we use `reduce` function the `urlValues` just set in 1 operation and I also think the null check `value[key] !== null ?` can be done as so `value[key] ?`. The second option just reduces noise and it also catches the exception when  `undefined` is returned as opposed to `null`